sealevel-ffi/src/config.rs

```diff
@@ -0,0 +1,130 @@
+use {crate::config::sealevel_config_opt::*, solana_rbpf::vm::Config, std::os::raw::c_int};
+
+/// Sealevel virtual machine config.
+#[derive(Default)]
+pub struct sealevel_config {
+    pub(crate) config: Config,
+    pub(crate) no_verify: bool,
+}
+
+#[repr(C)]
+pub enum sealevel_config_opt {
+    SEALEVEL_OPT_NONE,
+    SEALEVEL_OPT_NO_VERIFY,
+    SEALEVEL_OPT_MAX_CALL_DEPTH,
+    SEALEVEL_STACK_FRAME_SIZE,
+    SEALEVEL_ENABLE_STACK_FRAME_GAPS,
+    SEALEVEL_INSTRUCTION_METER_CHECKPOINT_DISTANCE,
+    SEALEVEL_ENABLE_INSTRUCTION_METER,
+    SEALEVEL_ENABLE_INSTRUCTION_TRACING,
+    SEALEVEL_ENABLE_SYMBOL_AND_SECTION_LABELS,
+    SEALEVEL_DISABLE_UNRESOLVED_SYMBOLS_AT_RUNTIME,
+    SEALEVEL_REJECT_BROKEN_ELFS,
+    SEALEVEL_NOOP_INSTRUCTION_RATIO,
+    SEALEVEL_SANITIZE_USER_PROVIDED_VALUES,
+    SEALEVEL_ENCRYPT_ENVIRONMENT_REGISTERS,
+    SEALEVEL_DISABLE_DEPRECATED_LOAD_INSTRUCTIONS,
+    SEALEVEL_SYSCALL_BPF_FUNCTION_HASH_COLLISION,
+    SEALEVEL_REJECT_CALLX_R10,
+    SEALEVEL_DYNAMIC_STACK_FRAMES,
+    SEALEVEL_ENABLE_SDIV,
+    SEALEVEL_OPTIMIZE_RODATA,
+    SEALEVEL_STATIC_SYSCALLS,
+    SEALEVEL_ENABLE_ELF_VADDR,
+}
+
+impl sealevel_config {
+    fn new() -> Self {
+        Self::default()
+    }
+}
+
+/// Creates a new Sealevel machine config.
+///
+/// # Safety
+/// Call `sealevel_config_free` on the return value after you are done using it.
+/// Failure to do so results in a memory leak.
+#[no_mangle]
+pub extern "C" fn sealevel_config_new() -> *mut sealevel_config {
+    let wrapper = sealevel_config::new();
+    Box::into_raw(Box::new(wrapper))
+}
+
+macro_rules! va_bool {
+    ($args:ident) => {
+        $args.arg::<c_int>() != 0
+    };
+}
+
+/// Sets a config option given the config key and exactly one value arg.
+///
+/// # Safety
+/// Avoid the following undefined behavior:
+/// - Passing the wrong argument type as the config value (each key documents the expected value).
+/// - Passing any other amount of function arguments than exactly 3.
+///
+/// cbindgen:ignore
+/// https://github.com/eqrion/cbindgen/issues/376
+#[no_mangle]
+pub unsafe extern "C" fn sealevel_config_setopt(
+    config: *mut sealevel_config,
+    key: sealevel_config_opt,
+    mut args: ...
```

Comment:
Variadic C args is supported on nightly only. 
`sealevel_config_setopt` accepts bool or usize depending on the config key, so that's why I used vararg here.
I need to check whether caller passing `bool` and callee expecting `usize` is sound behavior / Sys V ABI compliant.
If it is, then we don't need var arg here.

Reply:
strongly typed as `usize` for now